server/jsonHandler/recevieJSON.go

```diff
@@ -1,36 +1,55 @@
 package jsonHandler
 
 import (
-	"fmt"
+	dataStruct "devashishRaj/rpi_telemetry/server/dataStruct"
+	postgresDB "devashishRaj/rpi_telemetry/server/postgresDB"
+
+	//"fmt"
 	"log"
 	"net/http"
-	dataStruct "server/dataStruct"
-	postgresDB "server/postgresDB"
+	"sync"
 
 	"github.com/gin-gonic/gin"
 )
 
+var metricsMutex sync.Mutex
+var sysinfoMutex sync.Mutex
+
 func ReceiveJSON() {
-	var jsonData dataStruct.SystemInfo
+	var metricsData dataStruct.MetricsBatch
+	var sysinfoData dataStruct.SystemInfo
 
 	gin.SetMode(gin.ReleaseMode)
 
 	r := gin.Default()
 
-	r.POST("/rpi", func(c *gin.Context) {
-		if err := c.BindJSON(&jsonData); err != nil {
+	r.POST("/tele/metrics", func(c *gin.Context) {
+		// Todo(check plugin for fuzzy todo tag) add function to check
+		metricsMutex.Lock()
```

Comment:
metric writes shouldn't have a lock, as its high throughput op. imaging thousand of devices sending hundreds of metrics each every second. 
lock is anyway not needed now with conn pool.